docknetwork/dock-app
# DCKA 1477 - Dock wallet improvements on verification flow

## Proposed Changes
Implementation of some improvements to the Credential Presentation flow including:
- Making credentials selection checkbox respond to selection when any part of the card is clicked
- UI should differentiate between requested fields when sharing credential attributes
- Select a user's DID by default in the Identity selection stage. Select the first DID if the user has multiple DIDs.

## Testing Instructions
- To test the credential selection check box, 
   - Scan a credential verification request qr code
   - On the "Select the credentials to be presented" screen, select a credential by touching a credential card
   - Regardless of whether the checkbox icon is pressed, the card should be selected if any section of the card is pressed.
-  The UI changes made to differentiate requested fields from other fields is shown in the comparison below:
<img width="336" alt="Screenshot 2023-04-10 at 12 13 08 PM" src="https://user-images.githubusercontent.com/26460151/230891384-b1ea34ea-3588-476b-ad75-0625b4046fb0.png"> -> <img width="324" alt="Screenshot 2023-04-10 at 12 11 16 PM" src="https://user-images.githubusercontent.com/26460151/230891377-0a8cc221-f988-4044-b775-2100f9a0400c.png">
- To test the default DID selection, On the "Select Identity to present" screen, the user's DID should be selected by default.

## Additional Notes
- For the issue "When you have two credentials of the same type, the only way to tell them apart is by the issuance date - maybe tap to preview?", the changes made to the checkbox in this PR make it tricky to implement at the preview of the credentials. I think we could show credential ID to differentiate credentials instead. Are there any other suggestions for this?
- I am unable to reproduce the error “Credential not verified: a DID document lookup was successful, but the DID in question does not exist. This is different from a network error.” I'd love it if I can get a reproducible set of steps.
